autoPyTorch/pipeline/components/setup/network_backbone/ResNetBackbone.py

```diff
@@ -266,26 +327,34 @@ def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
             # if in_features != out_features
             # -> result = W_shortcut(A(BN(x))) + W_2(~D(A(BN(W_1(A(BN(x))))))
             x = self.start_norm(x)
-            residual = self.shortcut(x)
-
-        if self.config["use_shake_shake"]:
-            x1 = self.layers(x)
-            x2 = self.shake_shake_layers(x)
-            alpha, beta = shake_get_alpha_beta(self.training, x.is_cuda)
-            x = shake_shake(x1, x2, alpha, beta)
+            if self.config["use_skip_connection"]:
+                residual = self.shortcut(x)
+
+        # TODO make the below code better
+        if self.config["use_skip_connection"]:
+            if self.config["multi_branch_choice"] == 'shake-shake':
+                x1 = self.layers(x)
+                x2 = self.shake_shake_layers(x)
+                alpha, beta = shake_get_alpha_beta(self.training, x.is_cuda)
```

Comment:
If no definition of `device`, define device.
```suggestion
                alpha, beta = shake_get_alpha_beta(self.training, x.to(device))
```